opentracing-spring-messaging/src/main/java/io/opentracing/contrib/spring/integration/messaging/OpenTracingChannelInterceptor.java

```diff
@@ -76,23 +78,26 @@ public OpenTracingChannelInterceptor(Tracer tracer) {
       log.trace("Adding 'messageSent' header");
       carrier.put(Headers.MESSAGE_SENT_FROM_CLIENT, "true");
     }
+    log.trace(String.format("Pre-send: starting a new span %s , carrier extracted context %s", span, extractedContext));
 
     tracer.inject(span.context(), Format.Builtin.TEXT_MAP, carrier);
     return carrier.getMessage();
   }
 
   @Override
   public void afterSendCompletion(Message<?> message, MessageChannel channel, boolean sent, Exception ex) {
-    Scope scope = tracer.scopeManager().active();
-    if (scope == null) {
-      return;
+    if (message != null) {
```

Comment:
If a message is null there will be leaking scope - which should be always closed

Reply:
@pavolloffay Message cannot be null according to spring documentation. I have updated the already existing unit test in that direction.